app/services/employees/member_creator.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Employees
+  class MemberCreator < ApplicationService
+    attr_reader :member_params, :company
+
+    def initialize(params, company)
+      super()
+      @member_params = params
+      @company = company
+    end
+
+    def call
+      member = nil
```

Comment:
you can create on company in one line and no need in transaction.

Reply:
splitting the services is a good idea because we need the invite_resend feature
but when we create a new Employee the password field is empty. So User.invite!(params) provide the solition!
I propose this code (member_creator.rb):
```
    def call
      member = nil
      ActiveRecord::Base.transaction do
        member = Employees::MemberInviter.call(member_params)
        company.company_users.create!(user: member)
      end      
      member
    end
```

Reply:
I still would argue about this approach.

Reply:
Ok, my logic chain is:
- we need 2 insert operation on DB level: users then company_users (ORM generates them)
- it is important execute or rollback both operation together
- transaction block is needed
- member variable should be declare above for being visible from block outside (or @member)
Where is the mistake?

Reply:
you can return member from this line:
```
 member = Employees::MemberInviter.call(member_params)
```
And check if it exists add him:
```
 company.company_users.create!(user: member) if member
```
Not sure why we need to rollback

Reply:
And yes, we'll need transaction

Reply:
So the idea is to have one line for creation user and invite will be sent automatically.
1. create User object without saving
2. company.company_users.create!(user:u) - will create two records and send invite and rails will take care about transaction.